MSPApplicationSchema.json

```diff
@@ -1,1019 +1,1206 @@
+      "minLength":1,
+      "maxLength":25
     },
-    "PostalCodeType": {
-      "type": "string",
-      "minLength": 5,
-      "maxLength": 25
+    "PostalCodeType":{
+      "type":"string",
+      "minLength":5,
+      "maxLength":25
     },
-    "ProvinceOrStateType": {
-      "type": "string",
-      "minLength": 1,
-      "maxLength": 25
+    "ProvinceOrStateType":{
+      "type":"string",
+      "minLength":1,
+      "maxLength":25
     },
-    "CountryType": {
-      "type": "string",
-      "minLength": 1,
-      "maxLength": 30
+    "CountryType":{
+      "type":"string",
+      "minLength":1,
+      "maxLength":30
     },
-    "AddressType": {
-      "type": "object",
-      "properties": {
-        "addressLine1": {
-          "$ref": "#/definitions/SubAddressLineType"
+    "AddressType":{
+      "type":"object",
+      "properties":{
+        "addressLine1":{
+          "$ref":"#/definitions/SubAddressLineType"
         },
-        "addressLine2": {
-          "$ref": "#/definitions/SubAddressLineType"
+        "addressLine2":{
+          "$ref":"#/definitions/SubAddressLineType"
         },
-        "addressLine3": {
-          "$ref": "#/definitions/SubAddressLineType"
+        "addressLine3":{
+          "$ref":"#/definitions/SubAddressLineType"
         },
-        "city": {
-          "$ref": "#/definitions/CityType"
+        "city":{
+          "$ref":"#/definitions/CityType"
         },
-        "postalCode": {
-          "$ref": "#/definitions/PostalCodeType"
+        "postalCode":{
+          "$ref":"#/definitions/PostalCodeType"
         },
-        "provinceOrState": {
-          "$ref": "#/definitions/ProvinceOrStateType"
+        "provinceOrState":{
+          "$ref":"#/definitions/ProvinceOrStateType"
         },
-        "country": {
-          "$ref": "#/definitions/CountryType"
+        "country":{
+          "$ref":"#/definitions/CountryType"
         }
       },
-      "required": ["addressLine1", "city", "postalCode", "provinceOrState", "country"]
+      "required":[
+        "addressLine1",
+        "city",
+        "postalCode",
+        "provinceOrState",
+        "country"
+      ]
     },
-    "GenderType": {
-      "type": "string",
-      "pattern": "^[MF]$"
+    "GenderType":{
+      "type":"string",
+      "pattern":"^[MF]$"
     },
-    "PHNType": {
-      "type": "string",
-      "pattern": "^[1-9]([0-9]{9})$"
+    "PHNType":{
+      "type":"string",
+      "pattern":"^[1-9]([0-9]{9})$"
     },
-    "GroupNumberType": {
-      "type": "string",
-      "pattern": "^[0-9]*$"
+    "GroupNumberType":{
+      "type":"string",
+      "pattern":"^[0-9]*$"
     },
-    "YearType": {
-      "type": "string",
-      "pattern": "^[1-2]([0-9]{3})$"
+    "YearType":{
+      "type":"string",
+      "pattern":"^[1-2]([0-9]{3})$"
     },
-    "SINType": {
-      "type": "string",
-      "pattern": "^[1-9]([0-9]{8})$"
+    "SINType":{
+      "type":"string",
+      "pattern":"^[0-9]([0-9]{8})$"
```

Comment:
Can a SIN have a leading zero?

Reply:
I wondered about this too. I'll ask around to confirm; these changes came in from the attachment to the JIRA ticket.

Reply:
Did some research, and a leading zero is for fictitious SINs. As noted here: http://www.myfakeinfo.com/nationalidno/get-canada-sin.php

It might be good to include leading zeros for testing purposes.